Generator/Reader/XsdReader.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+namespace WsSys\DtoGeneratorBundle\Generator\Reader;
+
+use WsSys\DtoGeneratorBundle\Generator\Reader;
+use WsSys\DtoGeneratorBundle\Generator\DataMapper;
+
+/**
+ * Reads Xsd and it's elements
+ */
+class XsdReader
+{
+    /**
+     * @var \DOMDocument 
+     */
+    protected $dom;
+    
+    /**
+     * @var \DOMElement 
+     */
+    protected $start;
+    
+    /**
+     * Reads the XSD from given source
+     * @param string $source
+     */
+    public function read($source)
+    {
+        $this->dom = new \DOMDocument();
+        $this->dom->load($source);
+        $this->start = $this->dom->documentElement;
+    }
+    
+    /**
+     * Return First Element from Xsd
+     * 
+     * @return Element | null if not found
+     */
+    public function getFirstElementWithChildren()
+    {
+        $nodes = $this->start->childNodes;
+        foreach ($nodes as $node) {
+            if ($node->nodeType === XML_ELEMENT_NODE) {
+                $element = new Reader\Xsd\ComplexTypeElement();
+                $element->setName($node->getAttribute('name'))
+                        ->setElementAsFirst(true);
+                
+                $this->addChildrenElements($node, $element);
+
+                return $element;
+            }
+        }
+        return null;
+    }
+    
+    /**
+     * Add Children Elements to the given parentElement
+     * 
+     * @param DOMNode $parentNode
+     * @param Element $parentElement
+     */
+    protected function addChildrenElements($parentNode, &$parentElement) 
+    {
+        $children = $parentNode->childNodes;
+
+        foreach ($children as $node) {
+            if ($node->nodeType === XML_ELEMENT_NODE) {
+                switch ($node->localName) {
+                    case 'complexType':
```

Comment:
I would extract these cases into a Constant class for better portability of the code.